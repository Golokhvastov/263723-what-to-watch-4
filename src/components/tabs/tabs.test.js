import React from "react";
import renderer from "react-test-renderer";
import Tabs from "./tabs.jsx";

const mocks = {
  movie: {
    title: `Test1`,
    previewImage: `test1.jpg`,
    genre: `Test genre`,
    year: 1999,
    rating: {
      score: 5.5,
      votes: 110
    },
    descriptions: [
      `Test description1`,
      `Test description2`,
    ],
    director: `Test director`,
    starring: [
      `Test starring 1`,
      `Test starring 2`,
      `Test starring 3`,
      `Test starring 4`,
    ],
    runTime: 120,
    preview: `Test4`,
    reviews: [
      {
        rating: 7.9,
        date: `December 10, 2010`,
        author: `Test5`,
        text: `Test5`
      },
      {
        rating: 8.0,
        date: `December 20, 2020`,
        author: `Test6`,
        text: `Test6`
      },
      {
        rating: 9.0,
        date: `December 30, 2030`,
        author: `Test7`,
        text: `Test7`
      }
    ]
  }
};

describe(`Tabs render correctly`, () => {
  it(`Tabs with Overview`, () => {
    const tree = renderer
      .create(
          <Tabs
            movie={mocks.movie}
            activeItem = "Overview"
            onActiveItemChange = {() => {}}
          />
      ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it(`Tabs with Details`, () => {
    const tree = renderer
      .create(
          <Tabs
            movie={mocks.movie}
            activeItem = "Details"
            onActiveItemChange = {() => {}}
          />
      ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it(`Tabs with Reviews`, () => {
    const tree = renderer
      .create(
          <Tabs
            movie={mocks.movie}
            activeItem = "Reviews"
            onActiveItemChange = {() => {}}
          />
      ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
