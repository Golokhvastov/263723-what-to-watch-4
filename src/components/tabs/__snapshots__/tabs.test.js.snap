// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tabs render correctly Tabs with Details 1`] = `
<div
  className="movie-card__desc"
>
  <nav
    className="movie-nav movie-card__nav"
  >
    <ul
      className="movie-nav__list"
    >
      <li
        className="movie-nav__item "
      >
        <a
          className="movie-nav__link"
          href="#"
          onClick={[Function]}
        >
          Overview
        </a>
      </li>
      <li
        className="movie-nav__item movie-nav__item--active"
      >
        <a
          className="movie-nav__link"
          href="#"
          onClick={[Function]}
        >
          Details
        </a>
      </li>
      <li
        className="movie-nav__item "
      >
        <a
          className="movie-nav__link"
          href="#"
          onClick={[Function]}
        >
          Reviews
        </a>
      </li>
    </ul>
  </nav>
  <div
    className="movie-card__text movie-card__row"
  >
    <div
      className="movie-card__text-col"
    >
      <p
        className="movie-card__details-item"
      >
        <strong
          className="movie-card__details-name"
        >
          Director
        </strong>
        <span
          className="movie-card__details-value"
        >
          Test director
        </span>
      </p>
      <p
        className="movie-card__details-item"
      >
        <strong
          className="movie-card__details-name"
        >
          Starring
        </strong>
        <span
          className="movie-card__details-value"
        >
          Test starring 1
          ,
          <br />
          Test starring 2
          ,
          <br />
          Test starring 3
          ,
          <br />
          Test starring 4
        </span>
      </p>
    </div>
    <div
      className="movie-card__text-col"
    >
      <p
        className="movie-card__details-item"
      >
        <strong
          className="movie-card__details-name"
        >
          Run Time
        </strong>
        <span
          className="movie-card__details-value"
        >
          2h 0m
        </span>
      </p>
      <p
        className="movie-card__details-item"
      >
        <strong
          className="movie-card__details-name"
        >
          Genre
        </strong>
        <span
          className="movie-card__details-value"
        >
          Test genre
        </span>
      </p>
      <p
        className="movie-card__details-item"
      >
        <strong
          className="movie-card__details-name"
        >
          Released
        </strong>
        <span
          className="movie-card__details-value"
        >
          1999
        </span>
      </p>
    </div>
  </div>
</div>
`;

exports[`Tabs render correctly Tabs with Overview 1`] = `
<div
  className="movie-card__desc"
>
  <nav
    className="movie-nav movie-card__nav"
  >
    <ul
      className="movie-nav__list"
    >
      <li
        className="movie-nav__item movie-nav__item--active"
      >
        <a
          className="movie-nav__link"
          href="#"
          onClick={[Function]}
        >
          Overview
        </a>
      </li>
      <li
        className="movie-nav__item "
      >
        <a
          className="movie-nav__link"
          href="#"
          onClick={[Function]}
        >
          Details
        </a>
      </li>
      <li
        className="movie-nav__item "
      >
        <a
          className="movie-nav__link"
          href="#"
          onClick={[Function]}
        >
          Reviews
        </a>
      </li>
    </ul>
  </nav>
  <div
    className="movie-rating"
  >
    <div
      className="movie-rating__score"
    >
      5.5
    </div>
    <p
      className="movie-rating__meta"
    >
      <span
        className="movie-rating__level"
      >
        Good
      </span>
      <span
        className="movie-rating__count"
      >
        110
         ratings
      </span>
    </p>
  </div>
  <div
    className="movie-card__text"
  >
    <p>
      Test description1
    </p>
    <p>
      Test description2
    </p>
    <p
      className="movie-card__director"
    >
      <strong>
        Director: 
        Test director
      </strong>
    </p>
    <p
      className="movie-card__starring"
    >
      <strong>
        Starring: 
        Test starring 1, Test starring 2, Test starring 3, Test starring 4
         and other
      </strong>
    </p>
  </div>
</div>
`;

exports[`Tabs render correctly Tabs with Reviews 1`] = `
<div
  className="movie-card__desc"
>
  <nav
    className="movie-nav movie-card__nav"
  >
    <ul
      className="movie-nav__list"
    >
      <li
        className="movie-nav__item "
      >
        <a
          className="movie-nav__link"
          href="#"
          onClick={[Function]}
        >
          Overview
        </a>
      </li>
      <li
        className="movie-nav__item "
      >
        <a
          className="movie-nav__link"
          href="#"
          onClick={[Function]}
        >
          Details
        </a>
      </li>
      <li
        className="movie-nav__item movie-nav__item--active"
      >
        <a
          className="movie-nav__link"
          href="#"
          onClick={[Function]}
        >
          Reviews
        </a>
      </li>
    </ul>
  </nav>
  <div
    className="movie-card__reviews movie-card__row"
  >
    <div
      className="movie-card__reviews-col"
    >
      <div
        className="review"
      >
        <blockquote
          className="review__quote"
        >
          <p
            className="review__text"
          >
            Test5
          </p>
          <footer
            className="review__details"
          >
            <cite
              className="review__author"
            >
              Test5
            </cite>
            <time
              className="review__date"
              dateTime="2016-12-24"
            >
              December 10, 2010
            </time>
          </footer>
        </blockquote>
        <div
          className="review__rating"
        >
          7.9
        </div>
      </div>
      <div
        className="review"
      >
        <blockquote
          className="review__quote"
        >
          <p
            className="review__text"
          >
            Test6
          </p>
          <footer
            className="review__details"
          >
            <cite
              className="review__author"
            >
              Test6
            </cite>
            <time
              className="review__date"
              dateTime="2016-12-24"
            >
              December 20, 2020
            </time>
          </footer>
        </blockquote>
        <div
          className="review__rating"
        >
          8
        </div>
      </div>
    </div>
    <div
      className="movie-card__reviews-col"
    >
      <div
        className="review"
      >
        <blockquote
          className="review__quote"
        >
          <p
            className="review__text"
          >
            Test7
          </p>
          <footer
            className="review__details"
          >
            <cite
              className="review__author"
            >
              Test7
            </cite>
            <time
              className="review__date"
              dateTime="2016-12-24"
            >
              December 30, 2030
            </time>
          </footer>
        </blockquote>
        <div
          className="review__rating"
        >
          9
        </div>
      </div>
    </div>
  </div>
</div>
`;
